google/cloud/sql/connector/instance.py

```diff
@@ -79,7 +79,11 @@ def __init__(
     ) -> None:
         self.ip_addrs = ip_addrs
         self.database_version = database_version
-        self.context = ssl.SSLContext(ssl.PROTOCOL_TLS)
+        self.context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+
+        # update ssl.PROTOCOL_TLS_CLIENT defaults
+        self.context.check_hostname = False
+        self.context.verify_mode = ssl.VerifyMode.CERT_NONE
```

Comment:
I don't follow why this is CERT_NONE -- the docs say:

> In this mode, certificates are required from the other side of the socket connection; an [SSLError](https://docs.python.org/3/library/ssl.html#ssl.SSLError) will be raised if no certificate is provided, or if its validation fails.

I think we definitely want CERT_REQURIED?

Reply:
Yup we do want `CERT_REQUIRED`, updated the PR to only change `check_hostname`.